src/app/data-analytics-dashboard/oncology/oncology-reports/oncology-report-pdf-view/oncology-report-pdf.service.ts

```diff
@@ -0,0 +1,278 @@
+            }
+            ],
+            margin: [42, 20]
+          },
+          content: [{
+            stack: [
+              {
+                image: letterHead,
+                width: 150,
+                alignment: 'center'
+              },
+
+              {
+                text: this._formatReportIndicators(title) + ' Report',
+                style: 'mainHeader',
+                alignment: 'center'
+              }
+            ]
+          },
+          this.constructPdfSections(data, params)
+          ],
+          styles: {
+            header: {
+              fontSize: 14,
+              bold: true,
+              margin: [0, 0, 0, 10]
+            },
+            mainHeader: {
+              fontSize: 18,
+              bold: true,
+              margin: [0, 10, 0, 10]
+            },
+            subheader: {
+              fontSize: 10,
+              bold: true,
+              fillColor: '#979799',
+              margin: [0, 10, 0, 0]
+            },
+            headerstyle: {
+              fontSize: 10,
+              bold: true,
+              color: '#2a2a2a',
+              fillColor: '#d3d3d3'
+            },
+            defaultTable: {
+              fontSize: 10,
+              margin: [0, 0, 0, 5]
+            },
+            tableHeader: {
+              bold: true,
+              fontSize: 10,
+              color: 'black'
+            },
+            columns: {
+              fontSize: 11,
+              margin: [5, 2, 10, 20]
+            },
+            indicatorTitle: {
+              fontSize: 12,
+              bold: true,
+              margin: [0, 5, 0, 5]
+            }
+          },
+          defaultStyle: {
+            fontSize: 8
+          }
+        });
+      });
+    }).pipe(first());
+
+  }
+
+  private formatDate(date: Date) {
+    return _.isNull(date) ? 'None' : Moment(date).format('DD-MM-YYYY');
+  }
+
+  private getAppVersion(): string {
+    try {
+      return VERSION.version + VERSION.hash;
+    } catch (e) {
+      return '2';
+    }
+  }
+
+  private base64ToUint8Array(base64: any): Uint8Array {
+    const raw = atob(base64);
+    const uint8Array = new Uint8Array(raw.length);
+    for (let i = 0; i < raw.length; i++) {
+      uint8Array[i] = raw.charCodeAt(i);
+    }
+    return uint8Array;
+  }
+
+  private getLogo(url: string, callback: any): void {
+    const image: any = new Image();
+    image.onload = function () {
+      const canvas: any = document.createElement('canvas');
+      canvas.width = this.naturalWidth;
+      canvas.height = this.naturalHeight;
+
+      canvas.getContext('2d').drawImage(this, 0, 0);
+
+      callback(canvas.toDataURL('image/png'));
+
+      callback(canvas.toDataURL('image/png'));
+    };
+
+    image.src = url;
+  }
+
+  public constructPdfSections(data: Array<any>, params: any): Array<Array<any>> {
+    const sectionsArray: Array<Array<any>> = [[]];
+    let section: Array<any> = [];
+    data.forEach(element => {
```

Comment:
element is not a descriptive name. Choose a better name. something like **sectionRow** considering that you are creating a row for a section